src/sentry/search/events/fields.py

```diff
@@ -459,46 +460,58 @@ def format_column_arguments(column_args, arguments):
             column_args[i] = arguments[column_args[i].arg]
 
 
+def _lookback(columns, j, string):
+    """For parse_arguments, check that the current character is preceeded by string"""
+    if j < len(string):
+        return False
+    for i in range(len(string)):
+        if columns[i] != string[i]:
+            return False
+    return True
+
+
 def parse_arguments(function: str, columns: str) -> list[str]:
     """
     Some functions take a quoted string for their arguments that may contain commas,
     which requires special handling.
     This function attempts to be identical with the similarly named parse_arguments
     found in static/app/utils/discover/fields.tsx
     """
-    if (function != "to_other" and function != "count_if" and function != "spans_histogram") or len(
-        columns
-    ) == 0:
-        return [c.strip() for c in columns.split(",") if len(c.strip()) > 0]
-
     args = []
 
     quoted = False
+    in_tag = False
     escaped = False
 
     i, j = 0, 0
 
     while j < len(columns):
-        if i == j and columns[j] == '"':
+        if not in_tag and i == j and columns[j] == '"':
             # when we see a quote at the beginning of
             # an argument, then this is a quoted string
             quoted = True
+        elif not quoted and columns[j] == "[" and _lookback(columns, j, "tags"):
```

Comment:
cant you just hard code in the check with something simple like
```suggestion
        elif not quoted and columns[j] == "[" and columns[:j] == "tags":
```

Reply:
won't this only catch `thing(tags[foo])` but not `thing(blah, tags[foo])`?

Reply:
discussed in slack, current code only does the former lol. updating so it actually handles the latter